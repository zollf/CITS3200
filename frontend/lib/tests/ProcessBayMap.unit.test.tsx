import React from 'react';
import { TimeMocks } from '@/frontend/tests/mocks/time';
import { render } from '@testing-library/react';

import createListItems, { renderCleanedTimes } from '../ProcessBayMap';

const bayTimes = new Map();
TimeMocks.forEach((t) => bayTimes.set(t.slug, t));

describe('createListItems()', () => {
  it('works correctly given mock', () => {
    const listItems = createListItems(bayTimes);
    expect(listItems[0].map((t) => t.time)).toEqual(['08:00', '09:00']);
    expect(listItems[1].map((t) => t.time)).toEqual(['22:30', '23:30']);
    expect(listItems[1][1].endTime).toBe('00:00');
    expect(listItems[2].length).toBe(1);
  });
});

describe('renderCleanedTimes()', () => {
  it('matches its snapshot', () => {
    const component = () => (
      <table>
        <tbody>{renderCleanedTimes(bayTimes)}</tbody>
      </table>
    );
    expect(render(component()).baseElement).toMatchSnapshot();
  });

  it('matches its buffer snapshot', () => {
    const component = () => (
      <table>
        <tbody>{renderCleanedTimes(bayTimes, 30)}</tbody>
      </table>
    );
    expect(render(component()).baseElement).toMatchSnapshot();
  });
});
