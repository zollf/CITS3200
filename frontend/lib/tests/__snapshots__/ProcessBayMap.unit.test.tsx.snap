// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderCleanedTimes() matches its buffer snapshot 1`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr>
          <td>
            1
             
          </td>
          <td>
            08:00
            -
            09:30
             
            ± 30
          </td>
        </tr>
        <tr>
          <td>
            2
             
          </td>
          <td>
            22:30
            -
            00:00
             
            ± 30
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            00:00
            -
            00:30
             
            ± 30
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;

exports[`renderCleanedTimes() matches its snapshot 1`] = `
<body>
  <div>
    <table>
      <tbody>
        <tr>
          <td>
            1
             
          </td>
          <td>
            08:00
            -
            09:30
             
          </td>
        </tr>
        <tr>
          <td>
            2
             
          </td>
          <td>
            22:30
            -
            00:00
             
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            00:00
            -
            00:30
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;
